web/src/GlobalOptionsContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, PropsWithChildren, useContext } from "react"
+import { usePersistentState } from "./LocalStorage"
+
+export type GlobalOptions = {
```

Comment:
re: isn't everything a global option? this thought did cross my mind, too! in the spirit of the existing `sidebar_options`, i was thinking of `global-options` as global sorting and filter options. i can see the argument that the groups expanded state could also be encompassed by that, though my instinct is to leave group state separate for now, in case there are other group-level options/state we want to track, but we can definitely revisit in the future.

based on your feedback, i'm thinking it may be easier to keep the `sidebar_options` as-is in local storage, but extract it into a separate file and rename it `ResourceListOptions` (+ adding comments to watch for scope creep and noting that `alertsOnTop` is only used in the sidebar/detail view). i don't think it's too costly to revisit this structure, and i don't have a great idea of how bulk actions are going to influence this state, so it feels safer to not refactor too much prematurely.

Reply:
It's probably not worth a change right now, but an angle I'd forgotten that might be useful in the future - I suspect that persisting the resource name across runs is a bit weird and it would be better to put in sessionStorage than localStorage, and it's just in localStorage because it was added to SidebarOptions, which was already localStorage.

Reply:
> i don't think it's too costly to revisit this structure, and i don't have a great idea of how bulk actions are going to influence this state, so it feels safer to not refactor too much prematurely.

👍

Reply:
>  I suspect that persisting the resource name across runs is a bit weird and it would be better to put in sessionStorage than localStorage, and it's just in localStorage because it was added to SidebarOptions, which was already localStorage

ohh, that's good context! as a tilt user myself, i didn't think it was particularly intuitive for a search field to persist across sessions, but i thought maybe there was a reason why it was designed that way. i won't worry about it for this pr, but will keep in mind for the future.